certbot/client.py

```diff
@@ -52,9 +53,12 @@ def determine_user_agent(config):
     """
 
     if config.user_agent is None:
-        ua = "CertbotACMEClient/{0} ({1}) Authenticator/{2} Installer/{3}"
+        auto = "no" if cli.cli_command == "certbot" else "yes"
+        ua = ("CertbotACMEClient/{0} ({1}) Authenticator/{2} Installer/{3} Verb/{4} "
```

Comment:
The format for `User-Agent` strings is defined in [RFC 1945](https://www.ietf.org/rfc/rfc1945.txt) Section 10.15.

```
   The User-Agent request-header field contains information about the
   user agent originating the request. This is for statistical purposes,
   the tracing of protocol violations, and automated recognition of user
   agents for the sake of tailoring responses to avoid particular user
   agent limitations. Although it is not required, user agents should
   include this field with requests. The field can contain multiple
   product tokens (Section 3.7) and comments identifying the agent and
   any subproducts which form a significant part of the user agent. By
   convention, the product tokens are listed in order of their
   significance for identifying the application.

       User-Agent     = "User-Agent" ":" 1*( product | comment )
```

With `product` defined in Section 3.7.

```
   Product tokens are used to allow communicating applications to
   identify themselves via a simple product token, with an optional
   slash and version designator. Most fields using product tokens also
   allow subproducts which form a significant part of the application to
   be listed, separated by whitespace. By convention, the products are
   listed in order of their significance for identifying the
   application.

       product         = token ["/" product-version]
       product-version = token

   Examples:

       User-Agent: CERN-LineMode/2.15 libwww/2.17b3
```

And `comment` defined in Section 2.2.

```
   Comments may be included in some HTTP header fields by surrounding
   the comment text with parentheses. Comments are only allowed in
   fields containing "comment" as part of their field value definition.
   In all other fields, parentheses are considered part of the field
   value.

       comment        = "(" *( ctext | comment ) ")"
       ctext          = <any TEXT excluding "(" and ")">
```

While the authenticator and installer seem like they might the qualify as "subproducts", verb, renewing, and auto don't. It seems like comment(s) should be used for those instead.

For Python, we should look for existing conventions.